### PR TITLE
Move some templates to shared location

### LINKED ISSUE #5128: Code cleanup: The templates for react-native-windows init could share some files between C# and C++
In PR #5126 it was pointed out that there are files shared bewteen cpp and cs template. 
It would be great to update the code to reduce duplication.

### PR DESCRIPTION
**Pull request by dannyvv**
resolves #5128

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/5506)

### CODE REVIEW

**Comment by dannyvv** [comment]
Note that file `AutolinkedNativeModules.g.targets` is actually identical between cs and cpp as well, but that path is hardcoded in the script for autolinking and therefore moving it would break compatiblity.

**Comment by aeulitz** [comment]
> // AutolinkedNativeModules.g.cs contents generated by "react-native autolink-windows"

How come a C# file is added under a folder called "cpp/src"?

---
Refers to: vnext/local-cli/templates/cpp/src/AutolinkedNativeModules.g.cs:1 in 6206d25. [](commit_id = 6206d25dbf49a902b16e267b08e56622421bbeaf, deletion_comment = False)

File: vnext/local-cli/templates/shared/src/Package.appxmanifest [hunk 1]
```diff
@@ -28,7 +28,8 @@
   </Resources>
 
   <Applications>
-    <Application Id="App"
+    <Application
```
Comment (aeulitz) [inline vnext/local-cli/templates/shared/src/Package.appxmanifest:31]: >  [](start = 16, length = 1)

[nit] Trailing space.

File: vnext/local-cli/templates/shared/src/MainPage.xaml [hunk 2]
```diff
@@ -8,6 +8,14 @@
     xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" 
     mc:Ignorable="d"
     Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
-    <react:ReactRootView x:Name="reactRootView" ComponentName="{{ name }}"
-        Background="{ThemeResource ApplicationPageBackgroundThemeBrush}" Height="500"/>
+    <react:ReactRootView
```
Comment (aeulitz) [inline vnext/local-cli/templates/shared/src/MainPage.xaml:11]: >  [](start = 24, length = 1)

[nit, here and below] Trailing spaces.

**Comment by asklar** [comment]
I'd like to have @jonthysell review this if only as fyi to him :)

File: vnext/local-cli/tsconfig.json [hunk 3]
```diff
@@ -12,5 +12,5 @@
     // Not clean
     "strictFunctionTypes": false,
   },
-  "include": ["src"],
+  "include": ["src", "src/generator-windows/index.ts"],
```
Comment (NickGerleman) [inline vnext/local-cli/tsconfig.json:15]: I'm not sure this change does anything. Artifact from an older iteration?

Comment (dannyvv) [reply vnext/local-cli/tsconfig.json:15]: Missed reverting this file, something has been applying funny refactorings

**Comment by dannyvv** [comment]
> // AutolinkedNativeModules.g.cs contents generated by "react-native autolink-windows"

Shouldn't have been, 

---
In reply to: [656920494](https://github.com/microsoft/react-native-windows/pull/5506#issuecomment-656920494) [](ancestors = 656920494)

---
Refers to: vnext/local-cli/templates/cpp/src/AutolinkedNativeModules.g.cs:1 in 6206d25. [](commit_id = 6206d25dbf49a902b16e267b08e56622421bbeaf, deletion_comment = False)

**Comment by jonthysell** [comment]
> Note that file `AutolinkedNativeModules.g.targets` is actually identical between cs and cpp as well, but that path is hardcoded in the script for autolinking and therefore moving it would break compatiblity.

You could move it into a shared folder - all you'd have to do in the autolink script is update https://github.com/dannyvv/react-native-windows/blob/3aee6abfb9d438b596366031947f24b5452bdcce/vnext/local-cli/src/runWindows/utils/autolink.ts#L416 to `'shared'` instead of `projectLang`

**Comment by dannyvv** [comment]
> > Note that file `AutolinkedNativeModules.g.targets` is actually identical between cs and cpp as well, but that path is hardcoded in the script for autolinking and therefore moving it would break compatiblity.
> 
> You could move it into a shared folder - all you'd have to do in the autolink script is update https://github.com/dannyvv/react-native-windows/blob/3aee6abfb9d438b596366031947f24b5452bdcce/vnext/local-cli/src/runWindows/utils/autolink.ts#L416 to `'shared'` instead of `projectLang`

My concern was that we shipped this code in 0.62 so any updates might break it in 0.63..

**Pull request merged**